app/controllers/schedulers_controller.rb

```diff
@@ -119,4 +155,40 @@ def set_manage_scheduler_breadcrumb
 
     @breadcrumbs << (view_context.link_to "Manage Schedulers", course_schedulers_path(@course))
   end
+
+  def test_run_visual_scheduler(scheduler)
```

Comment:
like coderabbit is saying here, this is pretty much duplicated from `visual_run.` Can you refactor this so that both actions use some common code to do the visual run? Also, you should add some code that tries to compile the file first before doing the visual run and returning an error, see for example ` load_config_file` in `models/assessment.rb`

Reply:
Should be corrected with new commit.